addons/mrp_subcontracting/models/stock_move.py

```diff
@@ -247,8 +245,6 @@ def _update_subcontract_order_qty(self, new_quantity):
             productions = move.move_orig_ids.production_id.filtered(lambda p: p.state not in ('done', 'cancel'))[::-1]
             # Cancel productions until reach new_quantity
             for production in productions:
-                if quantity_to_remove <= 0.0:
-                    break
```

Comment:
Not sure why you're deleting that.
While I do think it's no longer necessary since the there should be only a single production for a subcontracted move. But if that's the case, then shouldn't you be deleting the `for` loop as well ?